ARCHITECTURE.md

```diff
@@ -1,292 +1,118 @@
-
-Interfaces:
-
--   READS from `Storage` while crawling, to check for differences (via the `diffing service`?)
--   WRITES to `Storage` when new versions are found
-
-<a id="org5cbd32e"></a>
-
-### Storage
-
-HTML and other resources are stored in Google cloud storage, along with an accompanying database that tracks changes. Database schemas are not currently known.
-
-Interfaces:
-
--   Communicates with `Page Collection` as above
--   Provides responses to `Diffing Service` queries about individual pages (see below)
--   Provides lists of pages and versions to `Researcher UI` (see below)
-
-<a id="orgba1a9c9"></a>
-
-### Diffing Service
-
-This service accepts queries about individual pages (and perhaps large sets of pages) and returns diffs in JSON format. We can use PageFreezer&rsquo;s existing service or create a replacement, depending on how closely our needs are matched by PageFreezer&rsquo;s. [We already have a candidate](https://github.com/edgi-govdata-archiving/differ) (developed to overcome some shortcomings of PageFreezer&rsquo;s *external* diff service), but we should probably evaluate the more robust internal diff UI with which we will soon be provided.
-
-Interfaces:
-
--   Queries `Storage` DB (via SQL?)
--   Provides JSON object to `UI`
-
-<a id="org0ee60b9"></a>
-
-### Filtration Service
-
-This service uses a set of heuristics to filter changes out of the result set presented in the UI. The heuristics will include generic and domain-specific rules. Eventually we would like to replace the ad-hoc rules in this service with machine-learning algorithms.
-
-Interfaces:
-
--   Either queries Storage for new pages or receives notification from Page Collection  when new pages are added to Storage.
--   Assigns a priortiy or other labels that can be used to filter important/unimportant changes and, ideally, prioritize them by estimated importance.
-
-<a id="org3b5f7b7"></a>
-
-### User Interface
-
-This web-based interface presents end users (analysts) with views of page collections (e.g., all pages from a given domain that pass the `filtration Service` tests, above) and individual pages. Analysts will be able to tag particular changes as belonging to a variety of categories, to be used later in EDGI&rsquo;s weekly reports. We&rsquo;ll place a high premium on ease of use, and speed of movement through changed pages; we&rsquo;ll probably want to be at least somewhat clever in our use of prefetching, etc.
-
-Interfaces:
-
--   Requests individual page diffs from `diff service`
--   Requests full lists of changed pages from `Storage`
--   Writes results of analysts&rsquo; inspection to `monitoring-database`
-
-<a id="org1c8db9d"></a>
+<a id="0"></a>
+## Definition of Terms
 
-### Monitoring Database
+* Page: a web page crawled over time by one or more services like the Internet
+  Archive or Versionista.
+* Version: a snapshot of a Page at a specific time (saved as HTML, for now).
```

Comment:
See above: not necessarily HTML. Also the thing that differentiates a “version” from a snapshot in our system is that it is necessarily *different* than the previous one. If we changed that, I think we’d want to change what we call this thing, too.

Reply:
If we change the terminology about calling "Page" -> "Snapshot", could we then say something like "Version: a Snapshot that is different from a previous Snapshot"?